libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/recenter/RecenterUIComponent.kt

```diff
@@ -42,7 +40,7 @@ internal class MapboxRecenterUIComponent(
     override val container: FrameLayout,
     private val view: MapboxRecenterButton,
     private val viewModel: RecenterViewModel,
-    private val lifecycleOwner: LifecycleOwner
+    private val lifecycleOwner: ViewLifecycleOwner
```

Comment:
I don't think we should require this. That defeats the purpose of the `LifecycleOwner` abstraction

We don't want to _require_ `RecenterUIComponent` to be owned by a view.

Reply:
Sure, that makes sense :+1: This was mostly done for access to `keepExecutingWhenStarted`. That can be an extension on `LifecycleOwner` though, it doesn't need to be `ViewLifecycleOwner`.

Reply:
Done, the only thing I'm worried about is that now we make the lifecycle of the `NavigationView` part of the public API. Not entirely sure that we want to do that but we have enough time to keep iterating and see if this might cause any problems.